### DISCUSSION TITLE
Domain is blocked [github.com]

### DISCUSSION THREAD

**Discussion by MNThomson** [badges/shields#7435] [Q&A] [answered]
Since a shield for the number of contributors to a GitHub Org doesn't exist (and it doesn't appear to be an easy addition), I've hacked together a script to output the number of unique contributors. This data is stored on GitHub, yet I cannot use it since the badge displays `Domain is Blocked`......I cannot find any explanation why GitHub is blocked for custom endpoints. Could somebody please elaborate?

Related Code: https://github.com/badges/shields/blob/23c0406bedfc6930735e8f5ea75dfe34faf1f290/services/endpoint/endpoint.service.js#L8 https://github.com/badges/shields/blob/23c0406bedfc6930735e8f5ea75dfe34faf1f290/services/endpoint/endpoint.service.js#L67-L69

**Comment by calebcartwright** [comment, marked as answer]
Because GitHub rate limits their APIs fairly aggressively, exceptionally so for unauthenticated requests. We deal with a fairly large volume of badge requests which means that we send a lot of API traffic at GitHub, and in order to avoid hitting the rate limits and having _all_ of our GitHub badges taken offline, we have to authenticate all those requests.

We can't start sending GitHub auth tokens from our pool to random user-defined endpoints, nor can we let someone "use" the endpoint badge to simply fetch content from GitHub anonymously directly from our servers as it would be a huge surface for abuse and DOS of our GitHub badges (which is the most used badge family we provide).

There's a number of viable, supported ways to get badges with content from GitHub

- Use one of our native GitHub badges (go to shields.io and search for 'github' via the search box at the top)
- Use our Dynamic badge with a raw.githubusercontent.com based url that allows using json path to query the badge message value
  - https://shields.io/#your-badge -> https://img.shields.io/badge/dynamic/json?color=blue&label=foo&query=%24.formation.web.quantity&url=https%3A%2F%2Fraw.githubusercontent.com%2Fbadges%2Fshields%2Fmaster%2Fapp.json
- Use the [Endpoint badge](https://shields.io/endpoint) in the intended model, where you actually deploy a custom endpoint (e.g. in a free service like runkit) and it is _your_ custom endpoint that makes API calls to GitHub, e.g. Shields server --> your endpoint (runkit) --> GitHub API

**Reply by MNThomson** [reply]
Awesome, thank you for the explanation. Makes total sense!
